sirepo/server.py

```diff
@@ -294,12 +294,24 @@ def app_find_by_name(simulation_type, application_mode, simulation_name):
 def app_import_file(simulation_type):
     f = flask.request.files['file']
     try:
-        srwblParam, appParam, el, pp = sirepo.sirepo_parser.sirepo_parser(f.read())
-        args = sirepo.sirepo_parser.list2dict(srwblParam)
-        v = sirepo.sirepo_parser.Struct(**args)
-        args = sirepo.sirepo_parser.list2dict(appParam)
-        app = sirepo.sirepo_parser.Struct(**args)
-        data = sirepo.sirepo_parser.parsed_dict(v, app, el, pp)
+        from werkzeug import secure_filename
+        filename = secure_filename(f.filename)
+        imported_file_path = os.path.join('./', filename)
+        try:
+            f.save(imported_file_path)
+        except:
+            raise Exception('File <{}> cannot be saved.'.format(filename))
```

Comment:
Do not catch the exception here. Rather just let it fall through to the outer exception. You'll have the arguments to generate a proper error message.